### DISCUSSION TITLE
How to add buttons to ActionRow?

### DISCUSSION THREAD

**Discussion by ghost** [nextcord/nextcord#1054] [Q&A] [answered]
```py
buttons_row = nextcord.ui.Row(left_button, right_button, cross_button)
message_component = nextcord.ui.MessageComponents(buttons_row)
await message.edit(components=message_component)
```
This doesnt work
```py
buttons_row = discord.ActionRow(left_button, right_button, cross_button)
await message.edit(components=[buttons_row])
```
This also doesnt work

**Comment by ooliver1** [comment]
Please refer to our [examples](https://github.com/nextcord/nextcord/tree/master/examples/views) and [documentation](https://docs.nextcord.dev/en/stable/api.html#bot-ui-kit) on buttons, this is nowhere near correct and I am not sure where you got those examples from.

**Reply by ghost** [reply]
@ooliver1 is there a way to add nextcord.ui.View to embeds? I can probably write buttons under that class but dont know how to add to embed. message.edit also doesn't seem to have components too

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/118899053/235372048-6f8835ab-a98f-4f29-bde7-76539654bf0f.png)
message.send doesnt seem to have components parts too. So I can add channel.send(embed=embed, file=nextcord.File(path)) but cant find a way to add buttons/components

**Reply by ooliver1** [reply, marked as answer]
Views and embeds are completely separate, you add views to messages and embeds to messages, they do not cross paths. `Message.edit` has a `view` parameter.

**Reply by ghost** [reply]
Maybe I have do view=view in message.channel.send?

**Reply by ooliver1** [reply]
> ![image](https://user-images.githubusercontent.com/118899053/235372048-6f8835ab-a98f-4f29-bde7-76539654bf0f.png) message.send doesnt seem to have components parts too. So I can add channel.send(embed=embed, file=nextcord.File(path)) but cant find a way to add buttons/components

As I referred to above, the examples and documentation never mention a `components` parameter.

**Reply by ghost** [reply]
> Views and embeds are completely separate, you add views to messages and embeds to messages, they do not cross paths. `Message.edit` has a `view` parameter.

Thanks. Just figured out it has view parameter. Will try and see if i get problem

**Reply by ghost** [reply]
Thank you. I have figured it out and successfully added Buttons.
